llama-index-integrations/llms/llama-index-llms-nvidia-ai-playground/llama_index/llms/nvidia_ai_playground/base.py

```diff
@@ -0,0 +1,311 @@
+""" NVIDIA AI Playground """
+
+from typing import Any, Callable, Dict, Optional, Sequence, cast, Awaitable
+
+from llama_index.core.base.llms.types import (
+    ChatMessage,
+    ChatResponse,
+    ChatResponseAsyncGen,
+    ChatResponseGen,
+    CompletionResponse,
+    CompletionResponseAsyncGen,
+    CompletionResponseGen,
+    LLMMetadata,
+    MessageRole,
+)
+from llama_index.core.bridge.pydantic import Field, PrivateAttr
+from llama_index.core.callbacks import CallbackManager
+from llama_index.core.constants import DEFAULT_TEMPERATURE
+from llama_index.core.llms.callbacks import (
+    llm_chat_callback,
+    llm_completion_callback,
+)
+from llama_index.core.base.llms.generic_utils import (
+    acompletion_to_chat_decorator,
+    achat_to_completion_decorator,
+    astream_chat_to_completion_decorator,
+    astream_completion_to_chat_decorator,
+    chat_to_completion_decorator,
+    get_from_param_or_env,
+    stream_chat_to_completion_decorator,
+)
+from llama_index.core.llms.llm import LLM
+from llama_index.core.types import BaseOutputParser, PydanticProgramMode
+
+from llama_index.llms.nvidia_ai_playground.utils import (
+    playground_modelname_to_contextsize,
+)
+
+from llama_index.llms.openai.utils import (
+    from_openai_message,
+    to_openai_message_dicts,
+)
+
+from openai import OpenAI as SyncOpenAI
+from openai import AsyncOpenAI
+
+from openai.types.chat.chat_completion_chunk import (
+    ChatCompletionChunk,
+    ChoiceDelta,
+)
+
+DEFAULT_PLAYGROUND_MODEL = "playground_nemotron_steerlm_8b"
+BASE_PLAYGROUND_URL = "https://integrate.api.nvidia.com/v1/"
+DEFAULT_PLAYGROUND_MAX_TOKENS = 512
+
+
+class NvidiaAIPlayground(LLM):
```

Comment:
chat completion models also support params -

- frequency_penalty: float -2..2 
- presence_penalty: float -2..2
- seed
- stop: str | list[str]

include them as properties or require passing as kwargs

in either case, make sure there's test coverage

Reply:
Passing as kwargs tested as working, added example to "test" notebook.